releases.md

```diff
@@ -83,7 +84,7 @@ As of 2018-11-16, PELUX 3.0 has been released! Documentation can be found
     - branch PELUX-3.0
     - tag v3.0
 
-* PELUX Software Factory Baseline
+* PELUX Development Handbook
     - URL: [https://github.com/Pelagicore/software-factory](https://github.com/Pelagicore/software-factory)
     - branch PELUX-3.0
     - tag v3.0
```

Comment:
The "Older releases" section hasn't been updated to point to the "releases" instead. I built this PR manually and could see some htmlproofer errors pointing to this.